Keep stdout stderr

lars-petter-hauge opened the pull request:
**Issue**
Continues the work on https://github.com/equinor/ert/issues/194

Does include commits from #520, but I'll rebase and fix everything.. The #520 will probably be merged first anyway.

**Approach**
First off, use `communicate` from `Popen` when using `PIPE` in order to collect and make sure that the program does not deadlock.

A bit unsure how the error should propagate upwards on this one, and who should be responsible for storing the result. I think I made the proper call on where to put it. 

This isn't complete, using `PIPE` also means that the stdout and stderr isn't sent to terminal, I'll fix that before it's merged so this doesn't change current behaviour

Commit:
Keep stdout and stderr when job is completed

joakim-hove commented on python/res/job_queue/ert_script.py:
```diff
@@ -19,6 +19,28 @@ def __init__(self, ert):
+        """ @rtype: str """
+        return self.__stdout
+
+    @stderr.setter
```
Having public read&write properties with names `stdout` and `stderr` is in my opinion a bit misleading, it is not `stdout` and `stderr`as such you are manipulating here after all, but rather the string content which has been written to the respective streams.

lars-petter-hauge replied:
yeh, agreed.

Did a quick test on how it feels when using the existing `defaultStackTrace` in `ErtScript`. That might be the most "how the rest works", but I think `ErtScript` still should keep the results and have it available.

Commit:
Use current functionality for sys

sys.exc.traceback has been deprecated since 1.5
sys.info() is also the only one of these functions that are python 3
compatible

Commit:
Expose job results from workflow_runner

Commit:
fixup! Keep stdout and stderr when job is completed

ManInFez reviewed (approved):
Good job!
